dist/ads.js

```diff
@@ -10,8 +10,13 @@ const fetchAndParse = async (url, parser) => {
   const { signal } = controller;
   setTimeout(() => controller.abort(), 5000);
 
+  // Google being popular ad-exchange hosting sellers.json at custom location, added its support
+  if (url.endsWith("google.com/sellers.json")) {
```

Comment:
`url` is actually a relative path ('/sellers.json'), so this condition is never satisfied.
You would need `document.location.origin == "https://www.google.com" && url == "/sellers.json"`

Also I see 2 more potential challenges:
- there may be an issue with cross-origin requests, as it has a redirect to another origin/domain.
- Google's file seems to weight 120Mb, so probably it would require more advanced processing (e.g. we're limiting request to 5sec).

Reply:
I don't think it's a common scenario, but we should consider those and identify them.
Then we could load files for analysis separately and build a complete graph.

Reply:
I'll incorporate your feedback, thanks!

Regarding the potential challenges you highlighted:

- Cross-origin request redirects will be handled by ```redirect: 'follow'```, no?
- So, when I test locally as well as on WPT Test, my original code itself works perfectly fine by extract >1million Google entries in 5secs. If you think it would be better, I can increase the timeout interval from 5secs to 10secs for request to fetch sellers.json for Google only.

Reply:
`redirect: follow` and `mode: cors` are a default setting, but it depends on the response CORS header whether the browser will allow access to it.

So we depend on server settings.
There's nothing we could adjust in fetch.